### PR TITLE
Host /unstable and /r0 versions of r0 APIs

### PR DESCRIPTION
**Pull request by illicitonion**

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -1,3 +1,18 @@
...
+  could take several hours for servers to start talking to ressurected servers,
+  even when they were receiving traffic from them (PR #393)
+* Don't advertise login token flow unless CAS is enabled. This caused issues
+  where some clients would always use the fallback API if they did not
+  recognize all login flows (PR #391)
+* Change /v2 sync API to rename ``private_user_data`` to ``account_data``
+  (PR #386)
+* Change /v2 sync API to remove the ``event_map`` and rename keys in ``rooms``
+  object (PR #389)
+
```
Comment (NegativeMjark) [inline CHANGES.rst]: I'm slightly confused as to why the change log update is appearing in this PR.

**Comment by NegativeMjark** [comment]
Other than the slightly curious changelog update it LGTM.

File: CHANGES.rst (hunk 1 above)
Comment (illicitonion) [inline CHANGES.rst]: I think I accidentally merged out from master :S

Undone, fixed
